server/service/src/programs/program_indicator/query.rs

```diff
@@ -153,32 +155,35 @@ impl ProgramIndicator {
 impl IndicatorLine {
     pub fn from_domain(line: IndicatorLineRow, columns: Vec<IndicatorColumnRow>) -> IndicatorLine {
         IndicatorLine {
-            name: line.description,
-            code: line.code,
+            name: line.description.clone(),
+            code: line.code.clone(),
             value: columns
                 .into_iter()
-                .map(|column| IndicatorColumn::from_domain(column))
+                .map(|column| IndicatorColumn::from_domain(column, line.id.clone()))
                 .collect(),
+            id: line.id,
         }
     }
 
     // TODO add to_domain utility function
 }
 
 impl IndicatorColumn {
-    pub fn from_domain(column: IndicatorColumnRow) -> IndicatorColumn {
+    pub fn from_domain(column: IndicatorColumnRow, line_id: String) -> IndicatorColumn {
         IndicatorColumn {
             header: column.header,
-            r#type: match column.value_type {
-                // TODO remove optional value type if we initialise default values on requisition creation?
-                Some(value_type) => match value_type {
-                    IndicatorValueType::String => ValueType::String,
-                    IndicatorValueType::Number => ValueType::Number,
-                },
-                None => ValueType::String,
-            },
+            // r#type: match column.value_type {
+            //     // TODO remove optional value type if we initialise default values on requisition creation?
+            //     Some(value_type) => match value_type {
+            //         IndicatorValueType::String => ValueType::String,
+            //         IndicatorValueType::Number => ValueType::Number,
+            //     },
+            //     None => ValueType::String,
+            // },
```

Comment:
I've left this commented as I'm not sure if we need it - and if we do need it, should it be here? Or should it be in the same node as the value?

I also wasn't sure if the value node should be more flattened rather than having a value node on the column.

You can see in IndicatorColumnNode, the value field returns another node which has within it the value table value and value.id.
 I had a feeling that leaving it nested as it is wasn't the best, but was reluctant to refactor too much in case flattening it also isn't correct.

I also wasn't quite sure how to take the loaded node result (which includes two values we want: id, and value), and then adding that two a flattened column value node